cartography/intel/crxcavator/crxcavator.py

```diff
@@ -103,6 +107,23 @@ def transform_extensions(extension_details):
             continue
         risk = data.get('risk', {})
         webstore = data.get('webstore', {})
+        manifest = data.get('manifest', {})
+        permissions = manifest.get('permissions', {})
+        for permission in permissions:
+            if type(permission) is dict:
+                parsed_permissions = parse_permissions_dict(permission)
+                permissions_set.update(parsed_permissions)
+                for sub_permission in parsed_permissions:
+                    extensions_permissions.append({
+                        'id': f"{extension_id}|{version}",
+                        'permission': sub_permission,
+                    })
+                continue
+            permissions_set.add(permission)
```

Comment:
If `permission` somehow happens to _not_ be a dict, then we still add it to `permissions_set`.

Later on in `load_extensions()` when this is `permissions_list`, if there is an object in `permissions_list` isn't a dict, will the sync crash?

I guess what I'm asking is to clarify the exact contract of the output objects from this function.  The doc of this function says that we "return a unique list of permissions", but can you please clarify what each permission object is expected to contain?

Reply:
@ecdavis can weigh in here, but I think each entry in the list should have the same datatype so that it will be easier to debug in the future.

Reply:
not sure what the issue is here? Each entry in the list will be a string after it is parsed out.

Reply:
I misread your comment earlier; got it.